### PR TITLE
Protect: Add TypeScript Support

### PR DESCRIPTION
**Pull request by nateweller**
## Proposed changes:

* Adds `typescript` as a development dependency
* Adds `tsconfig.json`
  * Based off of existing projects' configurations, i.e. [Boost](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/boost/tsconfig.json)

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Check out this branch
* Run `jetpack build plugins/protect`
* Ensure the plugin runs without regression.

### CODE REVIEW

**Comment by anomiex** [comment]
Looks fairly reasonable. It might be nice to use the same version of typescript as the rest of the monorepo to avoid confusion though. We could have Renovate update the typescript in rest of the monorepo if necessary.

> Includes an alias to support "absolute" imports, i.e. `import Foo from 'protect/components/foo'

Personally I've always found that sort of thing more confusing than helpful. Without knowing about the special configuration, that looks like it's a use of https://www.npmjs.com/package/protect. And once you start using it, you'll probably have to configure Webpack to know about it, and maybe Storybook, etc.

File: projects/plugins/protect/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+	"extends": "jetpack-js-tools/tsconfig.base.json",
+	"include": [ "./src/js" ],
+	"compilerOptions": {
+		"paths": {
+			"protect/*": [ "./src/js/*" ]
+		},
```
Comment (nateweller) [inline projects/plugins/protect/tsconfig.json:7]:
```suggestion
```

Comment (nateweller) [inline projects/plugins/protect/tsconfig.json]:
```suggestion
```

**Comment by nateweller** [comment]
Thanks for checking this out @anomiex!
* Removed the alias. Perhaps using a root alias like "$" or "@" would cause less confusion, but I just opted to leave it out for now.
* Changed the TS version to match the rest of the monorepo projects.

**Review by bhoop — DISMISSED** [review]
👍 , assuming the timeout errors aren't anything to worry about? (I'm not sure why this change would cause them)

**Comment by nateweller** [comment]
Timeouts are unrelated, but the required lock file test is failing now:

```
diff --git a/pnpm-lock.yaml b/pnpm-lock.yaml
index 4d08c2e..c3995b3 100644
--- a/pnpm-lock.yaml
+++ b/pnpm-lock.yaml
@@ -4070,7 +4070,7 @@ importers:
         version: 7.2.0
       '@wordpress/components':
         specifier: 28.2.0
-        version: 28.2.0(react-dom@18.3.1(react@18.3.1))(react@18.3.1)
+        version: 28.2.0(@types/react@18.3.1)(react-dom@18.3.1(react@18.3.1))(react@18.3.1)
       '@wordpress/data':
         specifier: 10.2.0
         version: 10.2.0(react@18.3.1)
---
Error: pnpm-lock.yaml is not up to date!
You can probably fix this by running `pnpm install` in the appropriate directory.
---
```

This may be due to introducing `@types/react` to the Protect plugin. The new lock file looks like I would expect it to be, but I bet @anomiex may know better? 😄 

Rebased trunk and ran `jetpack install -r` as well as `pnpm install` to be safe. No lock file changes came from either of those actions.

**Comment by anomiex** [comment]
Looks like this was one of those cases where a normal `pnpm install` was ok with it, but running `pnpm install --resolution-only` to force it to re-check stuff decided to make that change.

**Review by bhoop — APPROVED** [review]

**Pull request merged**